bridge-deploy/bridge_deploy/cli.py

```diff
@@ -0,0 +1,263 @@
+import click
+
+from deploy_tools.cli import (
+    jsonrpc_option,
+    keystore_option,
+    gas_option,
+    gas_price_option,
+    nonce_option,
+    auto_nonce_option,
+    connect_to_json_rpc,
+    retrieve_private_key,
+    get_nonce,
+)
+from deploy_tools.deploy import build_transaction_options
+from deploy_tools.files import validate_and_format_address, InvalidAddressException
+
+from bridge_deploy.home import (
+    deploy_home_block_reward_contract,
+    deploy_home_bridge_contract,
+    initialize_home_bridge_contract,
+)
+
+
+def validate_address(ctx, param, value):
+    """This function must be at the top of click commands using it"""
+    try:
+        return validate_and_format_address(value)
+    except InvalidAddressException as e:
+        raise click.BadParameter(
+            f"The address parameter is not recognized to be an address: {value}"
+        ) from e
+
+
+validator_set_address_option = click.option(
+    "--validator-set-address",
+    help=(
+        "The address of the bridge validator set contract or proxy contract"
+        ' that implements IBridgeValidators, "0x" prefixed string'
+    ),
+    type=str,
+    required=True,
+    callback=validate_address,
+    metavar="VALIDATOR_SET_ADDRESS",
+    envvar="VALIDATOR_SET_ADDRESS",
+)
+
+block_reward_address_option = click.option(
+    "--block-reward-address",
+    help='The address of the block reward contract, "0x" prefixed string',
+    type=str,
+    required=True,
+    callback=validate_address,
+    metavar="BLOCK_REWARD_ADDRESS",
+    envvar="BLOCK_REWARD_ADDRESS",
+)
+
+home_daily_limit_option = click.option(
+    "--home-daily-limit",
+    help="The daily transfer limit for the home bridge in WEI",
```

Comment:
can we convert this to ETH? Too easy to get the digits wrong if there are so many of them.

Reply:
I aggree with the difficulty, but if we go with ETH, we probably have to handle decimals. Is there a "fool proof" way to do this with click? @cducrest @schmir-at-bb

Reply:
probably along these lines:

```Py
from decimal import Decimal, InvalidOperation

def validate_eth_amount(ctx, param, value):
    try:
        eth_decimal = Decimal(value)
    except InvalidOperation:
        raise click.BadParameter(f"{value} is not a valid decimal number")
    else:
        wei_amount = eth_decimal * WEI_PER_ETH

    if int(wei_amount) != wei_amount:
        raise click.BadParameter(f"{value} contains fractional Wei component")

    return int(wei_amount)

@click.option(..., type=str, callback=validate_eth_amount, ...)
...
```

Reply:
Nice! Worked almost out of the box. There's still an issue with the Decimal conversion. Added a TODO.

Reply:
Updated.